Step-by-step-Tutorials/Custom-Storage-Providers.md

```diff
@@ -167,9 +168,9 @@ The format decisions have already been made, so coding up the `WriteStateAsync`
 
 
 ``` csharp
-public async Task WriteStateAsync(string grainType, Orleans.GrainReference grainRef, IGrainState grainState)
+public async Task WriteStateAsync(string grainType, GrainReference grainRef, IGrainState grainState)
 {
-    Dictionary<string, object> dataValues = grainState.AsDictionary();
+    IDictionary<string, object> dataValues = ((GrainState)grainState.State).AsDictionary();
     JavaScriptSerializer serializer = new JavaScriptSerializer();
```

Comment:
@jdom and @jason-bragg, the issue is fixed in 50262058512dccb724bb2883ddea8c0ac79c60a6. I did not mention ETag and kept it in its previous state.